packages/integrations/lit/server-shim.js

```diff
@@ -1,20 +1,28 @@
-import { installWindowOnGlobal } from '@lit-labs/ssr/lib/dom-shim.js';
+import {customElements as litCE} from '@lit-labs/ssr-dom-shim';
 
-if (typeof fetch === 'function') {
-	const _fetch = fetch;
-	installWindowOnGlobal();
-	globalThis.fetch = window.fetch = _fetch;
-} else {
-	installWindowOnGlobal();
+// Something at build time injects document.currentScript = undefined instead of
+// document.currentScript = null. This causes Sass build to fail because it
+// seems to be expecting `=== null`. This set to `undefined` doesn't seem to be
+// caused by Lit and only happens at build / test time, but not in dev or
+// preview time.
+if (globalThis.document) {
+	document.currentScript = null;
 }
 
-window.global = window;
-document.getElementsByTagName = () => [];
-// See https://github.com/lit/lit/issues/2393
-document.currentScript = null;
+// Astro seems to have a DOM shim and the only real difference that we need out
+// of the Lit DOM shim is that the Lit DOM shim does something that triggers
```

Comment:
Perhaps worth noting instead of "something" that we specifically need `observedAttributes` to be accessed. https://github.com/lit/lit/blob/a4eb5e60d393db3cd9d9a7f82c0c37457cd368eb/packages/labs/ssr-dom-shim/src/index.ts#L121-L123